### ISSUE TITLE
Could not find 'BlazorFluentUiBaseComponent.initializeFocusRects'

### ISSUE THREAD

**Issue by NiDeshp** [BlazorFluentUI/BlazorFluentUI#239]
I'm using BlazorFluentUI.AllComponents package with Blazor Client using .Net 5. I get following error when I try to use any BlazorFluentUI component like NavBar or DetailList.

I've followed steps mentioned in 
https://github.com/BlazorFluentUI/BlazorFluentUI/wiki/How-To-Use:-ClientSide-Blazor

crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Could not find 'BlazorFluentUiBaseComponent.initializeFocusRects' ('BlazorFluentUiBaseComponent' was undefined).
      Error: Could not find 'BlazorFluentUiBaseComponent.initializeFocusRects' ('BlazorFluentUiBaseComponent' was undefined).
          at https://localhost:5001/_framework/blazor.webassembly.js:1:1287
          at Array.forEach (<anonymous>)
          at e.findFunction (https://localhost:5001/_framework/blazor.webassembly.js:1:1247)
          at b (https://localhost:5001/_framework/blazor.webassembly.js:1:2989)
          at https://localhost:5001/_framework/blazor.webassembly.js:1:3935
          at new Promise (<anonymous>)
          at Object.beginInvokeJSFromDotNet (https://localhost:5001/_framework/blazor.webassembly.js:1:3908)
          at Object.w [as invokeJSFromDotNet] (https://localhost:5001/_framework/blazor.webassembly.js:1:64006)
          at _mono_wasm_invoke_js_blazor (https://localhost:5001/_framework/dotnet.5.0.0.js:1:190800)
          at do_icall (<anonymous>:wasm-function[10595]:0x194e46)

**Comment by limefrogyank** [comment]
Can you check that this file is actually loading in the dev console on your browser?
```
<script src="_content/BlazorFluentUI.BFUBaseComponent/blazorFluentUi.min.js"></script>
```
If that function can't be found, it's because this file is not loading...

**Comment by NiDeshp** [comment]
Thanks Lee, for quick response. Now the error has changed to 

TypeError: Failed to construct 'IntersectionObserver': The provided value is not of type '(Element or Document)'
    at new n (blazorFluentUi.min.js:formatted:1847)
    at Object.e.initialize (blazorFluentUi.min.js:formatted:1912)
    at blazor.webassembly.js:formatted:235
    at new Promise (<anonymous>)
    at Object.beginInvokeJSFromDotNet (blazor.webassembly.js:formatted:234)
    at Object.w [as invokeJSFromDotNet] (blazor.webassembly.js:formatted:3510)
    at _mono_wasm_invoke_js_blazor (dotnet.5.0.0.js:1)
    at do_icall (:5001/<anonymous>:wasm-function[10595]:0x194e46)
    at do_icall_wrapper (:5001/<anonymous>:wasm-function[3305]:0x79df6)
    at interp_exec_method (:5001/<anonymous>:wasm-function[2155]:0x44ad0)

![image](https://user-images.githubusercontent.com/16693402/99963922-5d872d00-2d47-11eb-96c2-5fb270895d59.png)

a.DotNet is undefined in call stack.

![image](https://user-images.githubusercontent.com/16693402/99964195-c8d0ff00-2d47-11eb-871c-b492accf1056.png)

**Comment by limefrogyank** [comment]
Maybe post a little more detail on how you're trying to use the `BFUList` or `BFUDetailsList`.  That's an error with the virtualizing part of the list.  If the `ItemsSource` is null, we've had problems with the whole thing crashing.  I pushed a fix, but I'm not sure we released a new nuget package for that yet.  If that's truly the problem in your case, you can try the nightly builds until we get a new version out.

**Comment by DerekChasse** [comment]
I'm experiencing the exact same issue as @NiDeshp.

This is a very simplified page which demonstrates the problem.  I noticed that there are breaking changes mentioned in the update to .Net5 but the remediation steps aren't obvious to apply.

```
@page "/Explore"

<h1>Explore</h1>

<BFUList ItemsSource=@data TItem="TestObject">
    <ItemTemplate>
        <div>
            <em>@context.Index</em>
            <span>@context.Item.Name</span>
            <span>@context.Item.Description</span>

        </div>
    </ItemTemplate>

</BFUList>


@code {
    class TestObject
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public TestObject(int id)
        {
            this.Id = id;
            this.Name = $"Name - {id}";
            this.Description = $"Description - {id}";
        }
    }

    IList<TestObject> data;

    protected override Task OnInitializedAsync()
    {
        data = new List<TestObject>();

        for (int i = 0; i < 20; i++)
        {
            data.Add(new TestObject(i));
        }

        return base.OnInitializedAsync();
    }
}
```

**Comment by limefrogyank** [comment]
> I'm experiencing the exact same issue as @NiDeshp.
> 
> This is a very simplified page which demonstrates the problem. I noticed that there are breaking changes mentioned in the update to .Net5 but the remediation steps aren't obvious to apply.


So this error is related to the changes with the list component.  `BFUList` (or anything that uses it) needs to find the element it can scroll in.  It must be wrapped by a `div` or other container element and needs the attribute `data-is-scrollable`.  This is functionally equivalent to making the `div` have a style of `overflow-y:auto`.  It doesn't need to be a direct element, but somewhere down the ancestor line.  

We should probably make this error not crash the system though and instead just fail to do scrolling. It would be nice to make this a separate issue though.

**Comment by DerekChasse** [comment]
OK, thanks for your assistance.  I'll give this a shot.

As a side not, your static site for the .NET 5 preview demo seems to be dead.

**Comment by limefrogyank** [comment]
Oh, I should remove the link then... it's not really in preview anymore.  Thanks

**Comment by Ralms** [comment]
I just followed the installation on a almost default Blazor WASM site and I'm also getting that error.
"Could not find 'BlazorFluentUiBaseComponent" 

```crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Could not find 'BlazorFluentUiBaseComponent.initializeFocusRects' ('BlazorFluentUiBaseComponent' was undefined).
      Error: Could not find 'BlazorFluentUiBaseComponent.initializeFocusRects' ('BlazorFluentUiBaseComponent' was undefined).
          at https://localhost:44308/_framework/blazor.webassembly.js:1:1287
          at Array.forEach (<anonymous>)
          at e.findFunction (https://localhost:44308/_framework/blazor.webassembly.js:1:1247)
          at b (https://localhost:44308/_framework/blazor.webassembly.js:1:2989)
          at https://localhost:44308/_framework/blazor.webassembly.js:1:3935
          at new Promise (<anonymous>)
          at Object.beginInvokeJSFromDotNet (https://localhost:44308/_framework/blazor.webassembly.js:1:3908)
          at Object.w [as invokeJSFromDotNet] (https://localhost:44308/_framework/blazor.webassembly.js:1:64218)
          at _mono_wasm_invoke_js_blazor (https://localhost:44308/_framework/dotnet.5.0.0.js:1:190800)
          at do_icall (<anonymous>:wasm-function[10595]:0x194e46)```

No Components are being used, the only thing has been the App.razor stuff from the guide `BFUTheme` and `BFULayerHost`. 
I'm not using anything else at this time, as such the mention regarding possibly being due to `BFUList` would not be applicable to me as I'm not using it. 

I have the line 
`<script src="_content/BlazorFluentUI.BFUBaseComponent/blazorFluentUi.min.js"></script>`
in my index.html and it seems to be loaded. 
Going to sources, localhost:44308, folder `_content`, I do have around 40 `BlazorFluentUI` components, including a `BlazorFluentUI.BFUBaseComponent`

Any further ideas? 

Thank you for the help.

**Comment by Ralms** [comment]
Ok, I've noticed that the error happens from something being called by `_framework/blazor.webassembly.js`.

It seems that moving the import in `index.html` from the head to the body stopped the error.
This is my current `index.html` body:
```
<body>
    <div id="app">Loading...</div>

    <div id="blazor-error-ui">
        An unhandled error has occurred.
        <a href="" class="reload">Reload</a>
        <a class="dismiss">🗙</a>
    </div>
    <script src="_content/BlazorFluentUI.BFUBaseComponent/blazorFluentUi.min.js"></script>
    <script src="_framework/blazor.webassembly.js"></script>
    <script src="_content/Microsoft.Authentication.WebAssembly.Msal/AuthenticationService.js"></script>
    <script>navigator.serviceWorker.register('service-worker.js');</script>
</body>
```

At first sight this seems to be working and have fixed the issue.

**Comment by limefrogyank** [comment]
Wow, this is why I love web development 🙄.  Sorry, I am not quite sure what is happening to your case.  If anyone can explain this, we can maybe fix it...

**Comment by Ralms** [comment]
> Wow, this is why I love web development 🙄. Sorry, I am not quite sure what is happening to your case. If anyone can explain this, we can maybe fix it...

Yeah I don't understand either, technically the head should load first.
I wonder if it was because when I did that change I also did a project clean and rebuild. 
I just tried moving it back to the head and continues to work. 
I will keep it in the body for performance reasons so that the loading HTML shows up as fast as possible, but this is really strange one.
